apps/admin_app/lib/admin_app_web/helpers.ex

```diff
@@ -62,4 +68,83 @@ defmodule AdminAppWeb.Helpers do
     |> Date.add(-1 * days)
     |> Date.to_string()
   end
+
+  def order_csv_exporter(user) do
+    path = "/tmp/orders.csv"
+    query = from(u in Order)
+
+    columns =
+      ~w(id number special_instructions billing_address shipping_address inserted_at updated_at user_id state)a
+
+    {:ok, file} =
+      Repo.transaction(fn ->
+        query
+        |> Repo.stream()
+        |> Stream.map(&parse_line/1)
+        |> CSV.encode(headers: columns, separator: ?\t, delimiter: "\n")
+        |> Enum.into(File.stream!(path, [:write, :utf8]))
+      end)
+
+    attachment = %Plug.Upload{
+      path: file.path,
+      content_type: "text/csv",
+      filename: "orders.csv"
+    }
+
+    OrderExportMail.order_export_mail(attachment, user, "csv")
+  end
+
+  defp parse_line(order) do
+    order |> Map.from_struct() |> parse_address()
+  end
+
+  defp parse_address(order) do
+    shipping_address = order.shipping_address |> format_address
+    billing_address = order.billing_address |> format_address
+    %{order | shipping_address: shipping_address, billing_address: billing_address}
+  end
+
+  defp format_address(address) do
+    case address do
+      nil ->
+        nil
+
+      address ->
+        address
+        |> Map.from_struct()
+        |> Enum.map(fn {key, value} -> value end)
+        |> Enum.join(" ")
+    end
+  end
+
+  def order_xlsx_exporter(user) do
+    orders = Repo.all(Order)
+
+    order_binary =
+      xlsx_generator(orders)
+      |> Elixlsx.write_to_memory("/tmp/orders.xlsx")
```

Comment:
Can we use `Elixlsx.write_to(%Workbook{}, filename)` for writing to file.

Reply:
Yes that would reduce a step to write the xlsx data on a file separately.